code/mob/living/critter/arena/fire_elemental.dm

```diff
@@ -67,3 +66,13 @@
 
 	get_disorient_protection_eye()
 		return(max(..(), 80))
+	death(var/gibbed)
+		..(gibbed, 0)
+		if (!gibbed)
+			playsound(src.loc, "sound/impact_sounds/burn_sizzle.ogg", 100, 1)
+			make_cleanable(/obj/decal/cleanable/ash,src.loc)
+			ghostize()
+			qdel(src)
+		else
+			playsound(src.loc, "sound/impact_sounds/burn_sizzle.ogg", 100, 1)
+			make_cleanable(/obj/decal/cleanable/ash,src.loc)
```

Comment:
This can be consolidated - the only difference is the ghost part.

Reply:
Fair enough. I'll probably make a pr to clean up the gunbot critter antag code, considering I based this on that code.